lib/ansible/modules/cloud/amazon/ec2_vpc_net_facts.py

```diff
@@ -57,77 +60,183 @@
+    vpc_ids = module.params.get('vpc_ids')
 
-def list_ec2_vpcs(connection, module):
+    # init empty list for return vars
+    vpc_info = list()
+    vpc_list = list()
 
-    filters = module.params.get("filters")
-    vpc_dict_array = []
+    # Get the basic VPC info
+    try:
+        response = connection.describe_vpcs(VpcIds=vpc_ids, Filters=filters)
+    except (ClientError, NoCredentialsError) as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Loop through results and create a list of VPC IDs
+    for vpc in response['Vpcs']:
+        vpc_list.append(vpc['VpcId'])
 
+    # We can get these results in bulk but still needs two separate calls to the API
     try:
-        all_vpcs = connection.get_all_vpcs(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
+        cl_enabled = connection.describe_vpc_classic_link(VpcIds=vpc_list)
+        cl_dns_support = connection.describe_vpc_classic_link_dns_support(VpcIds=vpc_list)
+    except (ClientError, NoCredentialsError) as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Loop through the results and add the other VPC attributes we gathered
+    for vpc in response['Vpcs']:
+        # We have to make two separate calls per VPC to get these attributes.
+        try:
+            dns_support = connection.describe_vpc_attribute(VpcId=vpc['VpcId'], Attribute='enableDnsSupport')
+            dns_hostnames = connection.describe_vpc_attribute(VpcId=vpc['VpcId'], Attribute='enableDnsHostnames')
+        except (ClientError, NoCredentialsError) as e:
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
 
-    for vpc in all_vpcs:
-        vpc_dict_array.append(get_vpc_info(vpc))
+        # loop through the ClassicLink Enabled results and add the value for the correct VPC
+        for item in cl_enabled['Vpcs']:
+            if vpc['VpcId'] == item['VpcId']:
+                vpc['ClassicLinkEnabled'] = item['ClassicLinkEnabled']
 
-    module.exit_json(vpcs=vpc_dict_array)
+        # loop through the ClassicLink DNS support results and add the value for the correct VPC
+        for item in cl_dns_support['Vpcs']:
+            if vpc['VpcId'] == item['VpcId']:
+                vpc['ClassicLinkDnsSupported'] = item['ClassicLinkDnsSupported']
 
+        # add the two DNS attributes
+        vpc['EnableDnsSupport'] = dns_support['EnableDnsSupport'].get('Value')
+        vpc['EnableDnsHostnames'] = dns_hostnames['EnableDnsHostnames'].get('Value')
+        # for backwards compatibility
+        vpc['id'] = vpc['VpcId']
+        vpc_info.append(camel_dict_to_snake_dict(vpc))
+        # convert tag list to ansible dict
+        vpc_info[-1]['tags'] = boto3_tag_list_to_ansible_dict(vpc['Tags'])
```

Comment:
Should be vpc_info[-1]['tags'] = boto3_tag_list_to_ansible_dict(vpc.get('Tags', [])) in case there are no tags